### ISSUE TITLE
The return type is incompatible

### ISSUE THREAD

**Issue by hsoehalim** [appium/java-client#323]
Hi There,

I'm new at this, perhaps I missed something during setup. I'm using eclipse Mars and jdk 1.7. After checking out the github project, I'm seeing a lot of compilation errors. In particulars "The return type is incompatible" on io.appium.java_client.AppiumDriver class. For example:

```
Multiple markers at this line
	- The return type is incompatible with WebDriver.findElements(By)
	- overrides 
	 io.appium.java_client.DefaultGenericMobileDriver<RequiredElementType>.findElements

@Override
public List<RequiredElementType> findElements(By by){
        return super.findElements(by);
}
```

![screen shot 2016-02-22 at 5 35 18 pm](https://cloud.githubusercontent.com/assets/1789810/13238827/aac24200-d98a-11e5-85c7-4b4261e76e33.png)

**Comment by TikhomirovSergey** [comment]
It is known issue. 
It is possible that this thing is a bug of javac compiler. Ecclipse compiler is required.

How to resolve:

- Install intellijIDEA. 

- import the maven project

- choose Eclipse compiler.
![](https://cloud.githubusercontent.com/assets/4927589/13251595/51a1cf48-da42-11e5-8af9-3691f685f1c4.png)

This problem will autamatically resolved if this PR is merged: https://github.com/SeleniumHQ/selenium/pull/863

**Comment by vikramvi** [comment]
@TikhomirovSergey  This issue is there with Eclipse IDE version 4.5.2. Please clarify how to resolve it ?

**Comment by SrinivasanTarget** [comment]
@vikramvi It will be resolved when #413 is merged.

**Comment by vikramvi** [comment]
Thanks @SrinivasanTarget , will wait for next release then.

**Comment by SrinivasanTarget** [comment]
@vikramvi There is a solution above, you can still use it.

**Comment by TikhomirovSergey** [comment]
Fixed at #413
